common/src/primitives/version.rs

```diff
@@ -14,7 +14,17 @@
 // limitations under the License.
 //
 // Author(s): A. Altonen
-use serialization::{Decode, Encode};
+use serialization::{Decode, Encode, Tagged};
+
+/// Version tag for SCALE-encoded values
+#[derive(Debug, Ord, PartialOrd, PartialEq, Eq, Encode, Decode, Copy, Clone, Tagged)]
+pub struct Version<const V: u8>(serialization::Tag<V>);
```

Comment:
As it stands now, I think this may be confusing. Can you please rename this to TagVersion, and move it to its own file tag_version.rs

The purpose here is to define a good name for this struct with its properties. Just calling it version will make discussions harder. We need a good name for it.

Later we'll also move SemVer to semver.rs

Reply:
Not sure about `TagVersion`. Maybe `VersionByte`? I find `VersionTag` slightly better if you feel tag should be there

Reply:
VersionTag is OK. I'd rather not put type information in the name.